Code/GraphMol/FileParsers/MolWriters.h

```diff
@@ -234,11 +238,13 @@ class RDKIT_FILEPARSERS_EXPORT TDTWriter : public MolWriter {
 
   //! \brief close our stream (the writer cannot be used again)
   void close() {
-    // if we've written any mols, finish with a "|" line
-    if (dp_ostream && d_molid > 0) {
-      *dp_ostream << "|\n";
+    if (dp_ostream) {
+      // if we've written any mols, finish with a "|" line
+      if (dp_ostream && d_molid > 0) {
```

Comment:
```suggestion
      if (d_molid > 0) {
```
`dp_ostream` has been checked to not be `nullptr` in the line above